src/components/Carousel.js

```diff
@@ -1,39 +1,190 @@
+    }
+  }
+
+  nextSlide = () => {
+    this.changeSlide(this.getCurrentValue() + this.props.slidesPerScroll);
+  }
+
+  prevSlide = () => {
+    this.changeSlide(this.getCurrentValue() - this.props.slidesPerScroll);
+  }
+
+
+  /* positioning */
+  getCurrentValue = () => {
+    const value = isNil(this.props.value) ? this.state.value : this.props.value;
+    return this.clamp(value);
+  }
+
+  getCarouselElementWidth = () => {
+    return this.state.carouselWidth / this.props.slidesPerPage;
+  }
+
+  getTransformOffset = () => {
+    return -(this.getCurrentValue() * this.getCarouselElementWidth());
+  }
+
+
+  /* rendering */
+  renderCarouselItems = () => {
+    const trackStyles = {
+      width: `${this.state.carouselWidth * this.props.children.length}px`,
+      transform: `translateX(${this.getTransformOffset()}px)`,
+    };
     return (
-      <div className={classnames('brainhub-carousel', this.props.additionalContainerClassName)}>
-        <button className="brainhub-carousel__arrows brainhub-carousel__arrow-left" type="button">Prev</button>
-        <ul className="brainhub-carousel__track">
-          {this.renderCarouselItems()}
-        </ul>
-        <button className="brainhub-carousel__arrows brainhub-carousel__arrow-right" type="button">Next</button>
-        <ul className="brainhub-carousel__dots">
-          {this.renderCarouselDots()}
+      <div className="BrainhubCarousel__trackContainer">
+        <ul className="BrainhubCarousel__track" style={trackStyles}>
+          {this.props.children.map((carouselItem, index) => (
+            <CarouselItem key={index} width={this.getCarouselElementWidth()}>
+              {carouselItem}
+            </CarouselItem>
+          ))}
         </ul>
       </div>
     );
   }
+
+  renderArrowWithAddedHandler = (element, onClick) => {
+    if (!element.props.onClick) {
+      return React.cloneElement(element, { onClick });
+    }
+    return element;
+  }
+
+  renderArrowLeft = () => {
+    if (this.props.arrowLeft) {
+      return this.renderArrowWithAddedHandler(this.props.arrowLeft, this.prevSlide);
+    }
+    if (this.props.arrows) {
+      return (
+        <div
+          className="BrainhubCarousel__arrows BrainhubCarousel__arrow-left"
+          type="button"
+          onClick={this.prevSlide}
+        >
+          &lt;
+        </div>
+      );
+    }
+    return null;
+  }
+
+  renderArrowRight = () => {
+    if (this.props.arrowRight) {
+      return this.renderArrowWithAddedHandler(this.props.arrowRight, this.nextSlide);
+    }
+    if (this.props.arrows) {
+      return (
+        <div
+          className="BrainhubCarousel__arrows BrainhubCarousel__arrow-right"
+          type="button"
+          onClick={this.nextSlide}
+        >
+          &gt;
+        </div>
+      );
+    }
+    return null;
+  }
+
+  renderCarouselDots = () => {
+    return (
+      <ul className="BrainhubCarousel__dots">
+        {this.props.children.map((carouselItem, index) => (
```

Comment:
Could you move this arrow function to a method?

Reply:
Removed in future PR